homeassistant/components/ecovacs/sensor.py

```diff
@@ -237,6 +274,32 @@ async def on_event(event: LifeSpanEvent) -> None:
         self._subscribe(self._capability.event, on_event)
 
 
+class EcovacsPositionSensor(
+    EcovacsDescriptionEntity[Capabilities, CapabilityEvent[PositionsEvent]],
+    SensorEntity,
+):
+    """Position sensor."""
+
+    entity_description: EcovacsPositionSensorEntityDescription
+
+    async def async_added_to_hass(self) -> None:
+        """Set up the event listeners now that hass is ready."""
+        await super().async_added_to_hass()
+
+        async def on_event(event: PositionsEvent) -> None:
+            for position in event.positions:
+                if position.type == self.entity_description.position_type:
+                    coordinates = {
+                        ATTRIBUTE_POSITION_X: position.x,
+                        ATTRIBUTE_POSITION_Y: position.y,
+                    }
+                    self._attr_native_value = json.dumps(coordinates)
+                    self._attr_extra_state_attributes = coordinates
```

Comment:
we want to avoid any extra attributes as far as possible, especial when they should just represent the entity state